<launch>

    <arg name="markerId"        default="5"/>
    <arg name="markerSize"      default="0.20"/>    <!-- in m -->
    <arg name="eye"             default="right"/>
    <arg name="marker_frame"    default="aruco_marker_frame"/>
    <!-- <arg name="marker_frame"    default="map"/> -->
    <arg name="ref_frame"       default=""/>  <!-- leave empty and the pose will be published wrt param parent_name -->
    <arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->

    <param name="/use_sim_time" value="false"/>

    <node name="republish_camera_image"
          pkg="image_transport"
          type="republish"
          args="_image_transport:=compressed in:=/Player0/camera/image raw out:=/Player0/camera/uncompressed"
          output="screen">
    </node>

    <node pkg="pose_transformer" type="new_transformer_node.py" name="transformer_node" output="screen"/>

    
    <node pkg="aruco_ros" type="single" name="aruco_drone">
        <remap from="/camera_info" to="/Player0/camera/camera_info/updated" />
        <remap from="/image" to="/Player0/camera/uncompressed" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>
        <param name="camera_frame"       value="HL2_camera"/>
        <param name="marker_frame"       value="$(arg marker_frame)" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node>

    <!-- <arg name="markerId"        default="5"/>
    <arg name="markerSize"      default="0.20"/> 
    <arg name="eye"             default="right"/>
    <arg name="marker_frame"    default="aruco_marker_frame"/>
    <arg name="marker_frame"    default="map"/>
    <arg name="ref_frame"       default=""/> 
    <arg name="corner_refinement" default="LINES" />

    <param name="/use_sim_time" value="false"/>

    <node pkg="aruco_ros" type="single" name="aruco_drone">
        <remap from="/camera_info" to="/camera/camera_info" />
        <remap from="/image" to="/camera/image_raw" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>
        <param name="camera_frame"       value="base_link"/>
        <param name="marker_frame"       value="$(arg marker_frame)" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node> -->


    <!-- <node pkg="pose_transformer" type="transformer_node.py" name="transformer_node" output="screen"/> -->

</launch>

